examples/asr/experimental/speech_to_text_aug.py

```diff
@@ -0,0 +1,243 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import random
+import subprocess
+from tempfile import NamedTemporaryFile
+
+import numpy as np
+import pytorch_lightning as pl
+import soundfile as sf
+from omegaconf import OmegaConf
+
+from nemo.collections.asr.models import EncDecCTCModel
+from nemo.collections.asr.parts import perturb
+from nemo.collections.asr.parts.segment import AudioSegment
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+
+
+class RirAndNoisePerturbation(perturb.Perturbation):
```

Comment:
Even if experimental, these two Perturbation classes should go directly into perturb.py, and not in this class. Model level training scripts should be as concise as possible. And putting them in this class prevents others from using these augmentations without editing or copy pasting code from this script, both of which are problematic.

Reply:
@titu1994 This class is just encapsulates calls to other perturbations. I felt this is not a fundamental perturbation like the other methods we have in perturb.py and that this may be a good example for users to create a custom perturbation. I don't mind moving this to perturb.py, let me know if you still think it should go there.

Reply:
The perturb() calls are not particularly trivial, so I believe they should he part of in built augmentations. Ease of use outside a particular script is valuable.